src/views/TodoView.vue

```diff
@@ -0,0 +1,329 @@
+<script setup lang="ts">
+import TableItem from "../components/TableItem.vue";
+import { computed, onMounted, ref } from "vue";
+import { ResponseData, TaskInfo, debounce } from "../util/shared";
+import {
+  deletePost,
+  getPosts,
+  postPost,
+  updatePost,
+} from "../util/HttpRequests";
+import FlexCard from "../components/FlexCard.vue";
+import FlexList from "../components/FlexList.vue";
+import FlexText from "../components/FlexText.vue";
+import AddTodo from "../components/AddTodo.vue"
+import { useRouter } from "vue-router";
+import IconButton from "../components/IconButton.vue";
+
+const DEBOUNCE_TIME = 250;
+
+let oldGet: ResponseData[]; // Keep a backup to restore from
+const router = useRouter()
+
+const sortingOrder = ref<SORT_OPTIONS>("task_dsc");
+
+const todoData = ref<ResponseData[]>([]);
+
+const filterInput = ref("");
+const addText = ref("");
+
+type SORT_OPTIONS = "task_asc" | "task_dsc" | "complete" | "incomplete";
+
+const sortingOptions: { text: string; value: SORT_OPTIONS }[] = [
+  { text: "Tasks (dsc)", value: "task_dsc" },
+  { text: "Tasks (asc)", value: "task_asc" },
+  { text: "Complete", value: "complete" },
+  { text: "Incomplete", value: "incomplete" },
+];
```

Comment:
Alternatively...
```typescript
const sortingOptions = [
  { text: "Tasks (dsc)", value: "task_dsc" },
  { text: "Tasks (asc)", value: "task_asc" },
  { text: "Complete", value: "complete" },
  { text: "Incomplete", value: "incomplete" },
] as const;
type SORT_OPTIONS = (typeof sortingOptions)[number]['value']
```
In your code the type is the source of truth and there will be an error if your actual options deviate from the type. In the alternative the options themselves are the source of truth and the type will automatically update whenever new options are added. Which is better depends on the situation.

Also types should ideally be formatted like you would a class name (i.e. SortOptions)

Reply:
I can see why using `['value']` extracts the values, but why does `[number]` convert it to a union type?

Reply:
[This should explain things](https://www.typescriptlang.org/docs/handbook/2/indexed-access-types.html)